lib/vrl/stdlib/src/add_datadog_tags.rs

```diff
@@ -0,0 +1,103 @@
+use std::borrow::Cow;
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct AddDatadogTags;
+
+impl Function for AddDatadogTags {
+    fn identifier(&self) -> &'static str {
+        "add_datadog_tags"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "value",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "tags",
+                kind: kind::ARRAY,
+                required: true,
+            },
+        ]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+        let tags = arguments.required("tags");
+
+        Ok(Box::new(AddDatadogTagsFn { value, tags }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "add datadog tags",
+            source: r#"add_datadog_tags!("env:beta,platform:windows", ["arch:amd64", "relay:vector"])"#,
+            result: Ok("arch:amd64,env:beta,platform:windows,relay:vector"),
+        }]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct AddDatadogTagsFn {
+    value: Box<dyn Expression>,
+    tags: Box<dyn Expression>,
+}
+
+impl Expression for AddDatadogTagsFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let new_tags = self.tags.resolve(ctx)?.try_array()?;
+        let mut new_tags_vec = new_tags
```

Comment:
```suggestion
        let mut new_tags = new_tags
```

In Rust, it's common practice to use variable shadowing, to avoid having to resort to more "awkward" variables names like here.